### DISCUSSION TITLE
AsyncNotifier build method dependency

### DISCUSSION THREAD

**Discussion by danielvillamizarGL** [rrousselGit/riverpod#2721] [Q&A]
I have the curren situation:

a. I have a FutureProvider that return a list of Items from the backend.
b. I have a AsyncNotifier that depends on previous FutureProvider

```dart
class TestNotifier extends AsyncNotifier<Item> {
  @override
  FutureOr<Item> build() {
    final itemsFutureProvider = ref.watch(itemsProvider);

    itemsFutureProvider.when(
      data: (item) {
        // Select the first item in the list
      },
       error: (error, stackTrace) => state = AsyncValue.error(error, stackTrace),
      loading: () => state = const AsyncValue.loading(),
    );
  }
}
``` 

The issue is that I want to control the loading and error state that comes from `itemsFutureProvider` and emit those in the `AsycnNotifier` but the `build` method prevents me from doing such a thing because I have to return the state from it.

The only workaround I found was to use `StateNotifier<AsyncValue<Item>>` and from the constructor, manually emit the states I need.

Is there a way to emit Loading state from `build` method from `AsyncNotifier` manually?

**Comment by AhmedLSayed9** [comment]
If you want to transit error/loading as is and only change data, you can use `.future`:
```dart
class TestNotifier extends AsyncNotifier<Item> {
  @override
  FutureOr<Item> build() {
    return ref.watch(itemsProvider.future).firstWhere((items)=> .. );
  }
}
```
